pkg/proxy/internal/util.go

```diff
@@ -46,8 +46,26 @@ func writeYaml(file string, data interface{}) error {
 	if err != nil {
 		return fmt.Errorf("failed to marshal yaml: %w", err)
 	}
-	if err := os.WriteFile(file, b, 0664); err != nil {
-		return fmt.Errorf("failed to write file: %w", err)
+
+	dir := filepath.Dir(file)
+	tmpFile, err := os.CreateTemp(dir, "tmp-*")
+	if err != nil {
+		return fmt.Errorf("failed to create temp file: %w", err)
 	}
+	defer os.Remove(tmpFile.Name())
+
+	if _, err := tmpFile.Write(b); err != nil {
+		tmpFile.Close()
+		return fmt.Errorf("failed to write to temp file: %w", err)
+	}
+
+	if err := tmpFile.Close(); err != nil {
```

Comment:
Why do we explicit close the temp file here and not just defer it?

Reply:
I thought it might help to make sure the data is flushed. TBH didn't do any research around this, just wanted to make sure we have successfully closed the file before renaming it. Do you think it's unnecessary? I can defer this to be consistent. WDYT?

Reply:
Did a test with deferring `tmpFile.Close()` and seems like the test passes and the write is successful. Using `defer`...

Reply:
Reverting back to the explicit `tmpFile.Close()` for safety measures.